### ISSUE TITLE
is_object is not detected as not null check for get_class

### ISSUE THREAD

**Issue by kesselb** [kalessil/phpinspectionsea#1583]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Extended), 4.0.4.1 |
| Language level | PHP 7.2                   |

Current behaviour
--

```
<?php

class A {
	/**
	 * @param object|null $event
	 * @return false|string
	 */
	public function getClassNameFromEvent(?object $event) {
		return is_object($event) ? get_class($event) : 'null';

	}
}

$obj = new A();
$obj->getClassNameFromEvent(null);
```

![image](https://user-images.githubusercontent.com/3902676/84016153-254c8780-a97d-11ea-8e46-47f6e47bbbc2.png)

Expected behaviour
--

No warning because `is_object(null)` is false. 

Thanks. Please keep the good work up :+1:

**Comment by pauljura** [comment]
Hi, I would also like to see this fixed.

I am seeing the same in my code, get_class should not have a warning if is_object() is true

![image](https://github.com/kalessil/phpinspectionsea/assets/42629973/dedfbf3d-af7a-4ac4-a443-e6378b91960f)

Thanks
